src/cascadia/TerminalSettingsEditor/ColorSchemes.cpp

```diff
@@ -233,75 +102,52 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
             {
                 if (const auto index{ tag.try_as<uint8_t>() })
                 {
-                    CurrentColorScheme().SetColorTableEntry(*index, newColor);
                     if (index < ColorTableDivider)
                     {
-                        _CurrentNonBrightColorTable.GetAt(*index).Color(newColor);
+                        _ViewModel.CurrentScheme().NonBrightColorTable().GetAt(*index).Color(newColor);
                     }
                     else
                     {
-                        _CurrentBrightColorTable.GetAt(*index - ColorTableDivider).Color(newColor);
+                        _ViewModel.CurrentScheme().BrightColorTable().GetAt(*index - ColorTableDivider).Color(newColor);
                     }
                 }
                 else if (const auto stringTag{ tag.try_as<hstring>() })
                 {
                     if (stringTag == ForegroundColorTag)
                     {
-                        CurrentColorScheme().Foreground(newColor);
-                        _CurrentForegroundColor.Color(newColor);
+                        _ViewModel.CurrentScheme().ForegroundColor().Color(newColor);
                     }
                     else if (stringTag == BackgroundColorTag)
                     {
-                        CurrentColorScheme().Background(newColor);
-                        _CurrentBackgroundColor.Color(newColor);
+                        _ViewModel.CurrentScheme().BackgroundColor().Color(newColor);
                     }
                     else if (stringTag == CursorColorTag)
                     {
-                        CurrentColorScheme().CursorColor(newColor);
-                        _CurrentCursorColor.Color(newColor);
+                        _ViewModel.CurrentScheme().CursorColor().Color(newColor);
                     }
                     else if (stringTag == SelectionBackgroundColorTag)
                     {
-                        CurrentColorScheme().SelectionBackground(newColor);
-                        _CurrentSelectionBackgroundColor.Color(newColor);
+                        _ViewModel.CurrentScheme().SelectionBackgroundColor().Color(newColor);
                     }
                 }
             }
         }
     }
 
-    bool ColorSchemes::CanDeleteCurrentScheme() const
-    {
-        if (const auto& scheme{ CurrentColorScheme() })
-        {
-            // Only allow this color scheme to be deleted if it's not provided in-box
-            const std::wstring myName{ scheme.Name() };
-            return std::find(std::begin(InBoxSchemes), std::end(InBoxSchemes), myName) == std::end(InBoxSchemes);
-        }
-        return false;
-    }
-
     void ColorSchemes::DeleteConfirmation_Click(const IInspectable& /*sender*/, const RoutedEventArgs& /*e*/)
```

Comment:
Same question as above

Reply:
The main reason is that we need to manipulate a bunch of the UI elements (like selecting the correct item in the combo box, hiding the delete button flyout, focusing the combo box), is it possible to give the ViewModel access to these?